app/routes.py

```diff
@@ -1146,44 +1193,46 @@ def add_harvest_record():
 
 ### Get record errors by record id
 @main.route("/harvest_record/<record_id>/errors", methods=["GET"])
+@valid_id_required
 def get_all_harvest_record_errors(record_id: str) -> list:
     try:
         record_errors = db.get_harvest_record_errors_by_record(record_id)
         return (
-            db._to_dict(record_errors)
-            if record_errors
-            else (STATUS_STRINGS_ENUM["404"], 404)
+            jsonify(db._to_dict(record_errors)) if record_errors else JSON_NOT_FOUND()
```

Comment:
Not sure this is right. There are perfectly valid records with no errors, that shouldn't return a 404. Looks like the db function always returns an empty array by default, so I'm not sure how to best handle this... Might require better handling in the upstream function to fully utilize properly.

Reply:
This replicates how it was working, so I'm not going to change it in this PR.